dartagnan/src/main/java/com/dat3m/dartagnan/analysis/graphRefinement/graphs/eventGraph/unary/TransitiveGraph.java

```diff
@@ -0,0 +1,82 @@
+package com.dat3m.dartagnan.analysis.graphRefinement.graphs.eventGraph.unary;
+
+import com.dat3m.dartagnan.analysis.graphRefinement.graphs.eventGraph.EventGraph;
+import com.dat3m.dartagnan.analysis.graphRefinement.graphs.eventGraph.utils.MaterializedGraph;
+import com.dat3m.dartagnan.analysis.graphRefinement.util.GraphVisitor;
+import com.dat3m.dartagnan.utils.collections.SetUtil;
+import com.dat3m.dartagnan.utils.timeable.Timestamp;
+import com.dat3m.dartagnan.verification.model.Edge;
+import com.dat3m.dartagnan.verification.model.ExecutionModel;
+
+import java.util.*;
+
+public class TransitiveGraph extends MaterializedGraph {
+
+    private final EventGraph inner;
+
+    @Override
+    public List<? extends EventGraph> getDependencies() {
+        return Collections.singletonList(inner);
+    }
+
+    public TransitiveGraph(EventGraph inner) {
+        this.inner = inner;
+    }
```

Comment:
This is common to all graphs in this folder and probably will remain like that for ever. Wouldn't be better to extract to a `AbstractUnaryGraph` class?

Reply:
The same reasons that apply to ```BinaryGraph``` apply here as well. There used to be a ```UnaryGraph``` class but its benefit was marginal and the disadvantages of wasting the single precious inheritance spot was too large. Hence, dependencies are only defined via an interface for maximal independance. Getting rid of those classes was one of the best design decision so I actually thought about doing the same for ```UnaryRelation``` and ```BinaryRelation```.